channels/internal/migrations/migrations.go

```diff
@@ -1,13 +1,111 @@
+	Sender peer.ID
+	// the party that is receiving the data (not who initiated the request)
+	Recipient peer.ID
+	// expected amount of data to be transferred
+	TotalSize uint64
+	// current status of this deal
+	Status datatransfer.Status
+	// total bytes read from this node and queued for sending (0 if receiver)
+	Queued uint64
+	// total bytes sent from this node (0 if receiver)
+	Sent uint64
+	// total bytes received by this node (0 if sender)
+	Received uint64
+	// more informative status on a channel
+	Message        string
+	Vouchers       []internal.EncodedVoucher
+	VoucherResults []internal.EncodedVoucherResult
+	// Number of blocks that have been received, including blocks that are
+	// present in more than one place in the DAG
+	ReceivedBlocksTotal int64
+	// Number of blocks that have been queued, including blocks that are
+	// present in more than one place in the DAG
+	QueuedBlocksTotal int64
+	// Number of blocks that have been sent, including blocks that are
+	// present in more than one place in the DAG
+	SentBlocksTotal int64
+	// DataLimit is the maximum data that can be transferred on this channel before
+	// revalidation. 0 indicates no limit.
+	DataLimit uint64
+	// RequiresFinalization indicates at the end of the transfer, the channel should
+	// be left open for a final settlement
+	RequiresFinalization bool
+	// Stages traces the execution fo a data transfer.
+	//
+	// EXPERIMENTAL; subject to change.
+	Stages *datatransfer.ChannelStages
+}
+
+func NoOpChannelState0To2(oldChannelState *ChannelStateV2) (*ChannelStateV2, error) {
+	return oldChannelState, nil
+}
+
+func MigrateChannelState2To3(oldChannelState *ChannelStateV2) (*internal.ChannelState, error) {
+	receivedIndex := basicnode.NewInt(oldChannelState.ReceivedBlocksTotal)
+	sentIndex := basicnode.NewInt(oldChannelState.SentBlocksTotal)
+	queuedIndex := basicnode.NewInt(oldChannelState.QueuedBlocksTotal)
+
+	return &internal.ChannelState{
+		SelfPeer:             oldChannelState.SelfPeer,
+		TransferID:           oldChannelState.TransferID,
+		Initiator:            oldChannelState.Initiator,
+		Responder:            oldChannelState.Responder,
+		BaseCid:              oldChannelState.BaseCid,
+		Selector:             oldChannelState.Selector,
+		Sender:               oldChannelState.Sender,
+		Recipient:            oldChannelState.Recipient,
+		TotalSize:            oldChannelState.TotalSize,
+		Status:               oldChannelState.Status,
+		Queued:               oldChannelState.Queued,
+		Sent:                 oldChannelState.Sent,
+		Received:             oldChannelState.Received,
+		Message:              oldChannelState.Message,
+		Vouchers:             oldChannelState.Vouchers,
+		VoucherResults:       oldChannelState.VoucherResults,
+		ReceivedIndex:        internal.CborGenCompatibleNode{Node: receivedIndex},
+		SentIndex:            internal.CborGenCompatibleNode{Node: sentIndex},
+		QueuedIndex:          internal.CborGenCompatibleNode{Node: queuedIndex},
```

Comment:
I imagine the thinking about adding migration testing is that this transition is pretty straightforward and the absolute hassle of having to come up with fixtures for this is just not worth it? It seems good from eyeballing it but we'll probably want to see it actually working in practice.

Reply:
no I should probably do that!